packages/button/src/styleSets.ts

```diff
@@ -0,0 +1,248 @@
+        },
+      },
+    },
+  },
+  [Variant.Primary]: {
+    text: uiColors.white,
+    border: '#158242',
+    background: {
+      color: {
+        base: uiColors.green.base,
+        active: uiColors.green.base,
+        'focus/hover': '#129f4c',
+      },
+      image: {
+        base: `linear-gradient(${uiColors.green.base}, #18964c)`,
+        active: `linear-gradient(#148040, #129f4c)`,
+        'focus/hover': `linear-gradient(#129f4c, #148040)`,
+      },
+      shadow: {
+        base: `inset 0 -1px 0 #158242`,
+        active: `inset 0 2px 2px ${uiColors.green.dark2}`,
+        'focus/hover': {
+          higher: `0 1px 4px ${transparentize(0.9, uiColors.black)}`,
+          lower: `inset 0 -1px 0 #158242`,
+        },
+      },
+    },
+  },
+  [Variant.Info]: {
+    text: uiColors.green.base,
+    border: uiColors.green.base,
+    background: {
+      color: {
+        base: 'transparent',
+        active: uiColors.green.base,
+        'focus/hover': '#129f4c',
+      },
+      image: {
+        base: 'none',
+        active: `linear-gradient(#148040, #129f4c)`,
+        'focus/hover': `linear-gradient(#129f4c, #148040)`,
+      },
+      shadow: {
+        base: '0 0 0 3px rgba(0, 0, 0, 0)', // Can't set to "none" because it may be set with other box shadows
+        active: `inset 0 2px 2px ${uiColors.green.dark2}`,
+        'focus/hover': {
+          higher: `0 1px 4px ${transparentize(0.9, uiColors.black)}`,
+          lower: `inset 0 -1px 0 #158242`,
+        },
+      },
+    },
+  },
+  [Variant.Danger]: {
+    text: uiColors.white,
+    border: uiColors.red.dark2,
+    background: {
+      color: {
+        base: uiColors.red.base,
+        active: uiColors.red.dark2,
+        'focus/hover': uiColors.red.dark2,
+      },
+      image: {
+        base: `linear-gradient(#e45b26, #b63016)`,
+        active: `linear-gradient(#ad231b, #e45b26)`,
+        'focus/hover': `linear-gradient(#e45b26, ${uiColors.red.dark2})`,
+      },
+      shadow: {
+        base: `inset 0 -1px 0 0 ${uiColors.red.dark2}`,
+        active: `inset 0 2px 2px ${uiColors.red.dark2}`,
+        'focus/hover': {
+          higher: `0 1px 4px ${transparentize(0.9, uiColors.black)}`,
+          lower: `inset 0 -1px 0 ${uiColors.red.dark2}`,
+        },
+      },
+    },
+  },
+  [Variant.Dark]: {
+    text: uiColors.white,
+    border: uiColors.gray.dark2,
+    background: {
+      color: {
+        base: uiColors.gray.dark1,
+        active: uiColors.gray.dark1,
+        'focus/hover': uiColors.gray.dark1,
+      },
+      image: {
+        base: `linear-gradient(${uiColors.gray.base}, ${uiColors.gray.dark1})`,
+        active: `linear-gradient(${uiColors.gray.dark1}, ${uiColors.gray.base})`,
+        'focus/hover': `linear-gradient(${uiColors.gray.base}, ${uiColors.gray.dark2})`,
+      },
+      shadow: {
+        base: `inset 0 -1px 0 ${uiColors.gray.dark2}`,
+        active: `inset 0 2px 2px ${uiColors.gray.dark2}`,
+        'focus/hover': {
```

Comment:
Using a box-shadow for the interaction ring is inconsistent with other applications in our codebase. This is important since we aren't transitioning the scale of the interaction ring here, just its transparency, so the animation is inconsistent, and less fluid. We should use the approach we have been to apply the interaction ring.

As an aside, the color set approach to variant and mode styling made finding how this style was applied, and where to leave this comment very hard. I'd prefer we take the approach I noted above to our styles, even if it's less typeable.

Reply:
>As an aside, the color set approach to variant and mode styling made finding how this style was applied, and where to leave this comment very hard. I'd prefer we take the approach I noted above to our styles, even if it's less typeable.

I'm curious as to why that was the case. I would've thought that the definition of `focusRing` would make these things straightforward to find:
https://github.com/mongodb/leafygreen-ui/pull/681/files#diff-fd57aaba57b850d63607350a5dd87f40dff04d2eff66a61ea5b9c363c93d26d6R106
```